eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/UtxosSpec.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Cardano.Ledger.Conway.Imp.UtxosSpec where
+
+import Cardano.Ledger.Address (Addr (..))
+import Cardano.Ledger.BaseTypes (Inject (..), Network (..), StrictMaybe (..), maybeToStrictMaybe)
+import Cardano.Ledger.Coin (Coin (..))
+import Cardano.Ledger.Conway.Core (
+  AlonzoEraScript (..),
+  AlonzoEraTxOut (..),
+  BabbageEraTxBody (..),
+  BabbageEraTxOut (..),
+  EraTx (..),
+  EraTxBody (..),
+  EraTxOut (..),
+  txIdTx,
+ )
+import Cardano.Ledger.Credential (Credential (..), StakeReference (..))
+import Cardano.Ledger.Plutus (Data (..), hashData)
+import Cardano.Ledger.TxIn (mkTxInPartial)
+import qualified Data.Sequence.Strict as SSeq
+import qualified Data.Set as Set
+import Lens.Micro ((&), (.~))
+import qualified PlutusLedgerApi.V1 as P1
+import Test.Cardano.Ledger.Conway.ImpTest (ImpTestState, ShelleyEraImp, freshKeyHash, impAddPlutusScript, lookupKeyPair, submitTxAnn, submitTxAnn_)
+import Test.Cardano.Ledger.Core.KeyPair (mkAddr)
+import Test.Cardano.Ledger.Imp.Common
+import Test.Cardano.Ledger.Plutus (PlutusArgs (..), ScriptTestContext (..))
+import Test.Cardano.Ledger.Plutus.Examples (guessTheNumber3)
+
+spec ::
+  forall era.
+  ( ShelleyEraImp era
+  , BabbageEraTxBody era
+  ) =>
+  SpecWith (ImpTestState era)
+spec =
+  describe "UTXOS" $ do
+    it "can use reference scripts" $ do
```

Comment:
Could you please ad one more test that checks that validation fails when a transaction tries to use a TxOut that besides PlutusV1 reference script also contains an inline datum. Since those should still result in a failure.